src/context.rs

```diff
@@ -26,6 +28,79 @@ impl<'a, 'b, 'c, 'info, T: Accounts<'info>> Context<'a, 'b, 'c, 'info, T> {
     }
 }
 
+/// Context speciying non-argument inputs for cross-program-invocations
+/// targeted at program state.
+pub struct StateCpiContext<'a, 'b, 'c, 'info, T: Accounts<'info>> {
```

Comment:
TODO: it would be convenient to have `CpiState` version of https://docs.rs/anchor-lang/0.4.2/anchor_lang/struct.ProgramState.html (similar to `CpiAccount`), that had a method to build a `StateCpiContext` on it.